code/modules/mod/modules/modules_timeline.dm

```diff
@@ -0,0 +1,448 @@
+ * Unlinks a chrono field if it is connected to a tem already.
+ *
+ * Arguments:
+ * * field: chronofield we are attempting to link to this module.
+ */
+/obj/item/mod/module/tem/proc/field_connect(obj/structure/chrono_field/field)
+	if(field.tem)
+		if(field.captured)
+			to_chat(mod.wearer, span_alert("<b>FAIL: <i>[field.captured]</i> already has an existing connection.</b>"))
+		field_disconnect(field)
+		return
+	startpos = get_turf(mod.wearer)
+	field = field
+	field.tem = src
+	if(field.captured)
+		to_chat(mod.wearer, span_notice("Connection established with target: <b>[field.captured]</b>"))
+
+/**
+ * ### field_disconnect
+ *
+ * Unlinks a chrono field from this module.
+ *
+ * Arguments:
+ * * field: chronofield we are attempting to unlink from this module.
+ */
+/obj/item/mod/module/tem/proc/field_disconnect(obj/structure/chrono_field/field)
+	if(field && field == field)
+		if(field.tem == src)
+			field.tem = null
+		if(field.captured)
+			to_chat(mod.wearer, span_alert("Disconnected from target: <b>[field.captured]</b>"))
+	field = null
+	startpos = null
+
+/**
+ * ### field_check
+ *
+ * Checks to see if  our field can still be linked to the tem. If it isn't, it will unlink the field.
+ *
+ * Arguments:
+ * * field: chronofield we're checking the connection's validity on.
+ */
+/obj/item/mod/module/tem/proc/field_check(obj/structure/chrono_field/field)
+	if(!field)
+		return FALSE
+	if(field == field)
+		var/turf/currentpos = get_turf(src)
+		if(currentpos == startpos && mod.wearer.body_position == STANDING_UP && !HAS_TRAIT(mod.wearer, TRAIT_INCAPACITATED) && (field in view(CHRONO_BEAM_RANGE, currentpos)))
+			return TRUE
+	field_disconnect(field)
+	return FALSE
+
+/obj/projectile/energy/chrono_beam
+	name = "eradication beam"
+	icon_state = "chronobolt"
+	range = CHRONO_BEAM_RANGE
+	nodamage = TRUE
+	///reference to the tem... given by the tem! weakref because back in the day we didn't know about harddels- or maybe we didn't care.
+	var/datum/weakref/tem_weakref
+
+/obj/projectile/energy/chrono_beam/on_hit(atom/target)
+	var/obj/item/mod/module/tem/tem = tem_weakref.resolve()
+	if(target && tem && isliving(target))
+		var/obj/structure/chrono_field/field = new(target.loc, target, tem)
+		tem.field_connect(field)
+
+/obj/structure/chrono_field
+	name = "eradication field"
+	desc = "An aura of time-bluespace energy."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "chronofield"
+	density = FALSE
+	anchored = TRUE
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	move_resist = INFINITY
+	interaction_flags_atom = NONE
+	///mob being eradicated by this field
+	var/mob/living/captured
+	///linked module. while this exists, the field will progress towards eradication. while it isn't, the field progresses away until it disappears. see attached for a special case
+	var/obj/item/mod/module/tem/tem
```

Comment:
These need to be handled in `Destroy()`?

Reply:
it is nulled when field disconnect is called in destroy, but i'm going to rewrite it to make more sense.